xapian-applications/omega/docs/omegascript.rst

```diff
@@ -447,6 +447,15 @@ $queryterms
 $range{START,END}
 	return list of values between ``START`` and ``END``.
 
+$random
+
+$random{HIGH}
+
+$random{LOW,HIGH}
+	return a random value in the range [``LOW``, ``HIGH``).
+	If the value of ``LOW`` is ommitted, then the range [0, ``HIGH``) is used.
+	If both the values are ommitted, then the range [0, RAND_MAX] is used.
```

Comment:
"ommitted" should be "omitted" (twice).

The ticket actually proposed returning a real value, though I think that's less useful than an integer as OmegaScript doesn't really provide a way to work with real numbers.  I think the motivation for this was to enable things like A/B testing where a different version is shown to a proportion of users, and for that integer random numbers should be sufficient.

I'm not sure the no-parameter version is very useful - it returns a random number up to some platform-dependent value which you can't access from OmegaScript.  I struggle to see how I would use that.

I'd probably just provide the single-parameter version for now - `$random{2,10}` is the same as `$add{$random{8},2}` and the inclusive/exclusive bounds seem potentially confusing.

Reply:
Changed $random in the latest commit to support exactly one argument.